frontend/src/home/components/Homepage.jsx

```diff
@@ -25,26 +25,42 @@ export default class Homepage extends Component {
     }
 
     static propTypes = {
-        dispatch: PropTypes.func.isRequired,
         onChangeLocation: PropTypes.func.isRequired,
         showOnboarding: PropTypes.bool.isRequired,
-        user: PropTypes.object.isRequired
+        user: PropTypes.object.isRequired,
+        activity: PropTypes.array,
+        hasSampleDataset: PropTypes.bool,
+        fetchActivity: PropTypes.func.isRequired,
+        fetchRecentViews: PropTypes.func.isRequired,
+        fetchHasSampleDataset: PropTypes.func.isRequired
     };
 
+    async componentDidMount() {
+        try {
+            await this.props.fetchHasSampleDataset();
```

Comment:
There are a lot of places we reload data more often than we should. That doesn't mean this is a good thing, I just think we need to think about it on a broader scale at some point (soonish)